providers.tf

```diff
@@ -1,16 +1,28 @@
 # This is the default provider that is used to create resources inside
-# the Users account
+# the Users account.
 provider "aws" {
 }
 
+# This is the provider that is used to create the role that can be
+# assumed to perform CI functions.
```

Comment:
...create the role **in the Images Production account** that can be assumed...

Reply:
For the AMI build user the production and staging accounts will definitely be the Images accounts, and the same is true most of the time for Molecule CI test users since they will usually be grabbing parameters from SSM Parameter Store.  Consider, though, a Molecule CI test user that pulls an object from a bucket that _does not_ live inside the Images accounts.  This Terraform module supports such a case, and that is why I purposefully left out references to "Images production" and "Images staging."

Reply:
One improvement to consider is simply create a user and attaching policies to it specifying roles that can be assumed - possibly across multiple accounts.  Then we would pass the role to be assumed to `converge.yml` and use the [`sts_assume_role`](https://docs.ansible.com/ansible/latest/modules/sts_assume_role_module.html) Ansible module to assume the desired role, pull down the SSM Parameter Store parameter value (for example), and pass it as an input to the Ansible module being tested.  This sort of setup would:
* Be simpler, in my opinion
* Be more flexible, since one could assume roles in more than one account
* Have the drawback that one would have to specify hardcoded role ARNs in `converge.yml`, or as extra parameters to add to the Ansible command line.

@felddy @dav3r @mcdonnnj @hillaryj I'm curious what others think of this approach.

Reply:
I'm not a huge fan of ARNs in the repos.   I like the idea otherwise.   Is there another way to get the creds to molecule?

Reply:
I will merge this PR, but we can continue this discussion here.  Or elsewhere, if folks prefer.